scripts/flattener.py

```diff
@@ -1420,6 +1420,18 @@ def main(mfinal_id):
 		get_sex_ontology(df)
 		cxg_obs = pd.merge(cxg_obs, df[['disease_ontology_term_id', 'reported_diseases', 'sex_ontology_term_id']], left_on="raw_matrix_accession", right_index=True, how="left" )
 
+	# Check that primary_portion.obs_field of ProcessedMatrixFile is present in cxg_obs
+	primary_portion = mfinal_obj.get('primary_portion')
```

Comment:
`primary_portion` is not found in all objects, and will error with, for example if you flatten LATDF512YHM

Reply:
Can be changed to` primary_portion = mfinal_obj.get('primary_portion', None)`. This returns None if there is no such field.